website/sampling-consensus.md

```diff
@@ -1,37 +1,79 @@
 
-As previously mentioned, the Atoma protocol can enforce determinism for
-execution workloads (a.k.a requests) by compute replication, through selecting uniformly at random a given number of nodes to run the same compute. 
+The issue with BFT consensus is that the large majority of nodes (usually around 2/3rds) must agree on a transaction or block of transactions.
 
-Depending on the number of sampled nodes, different levels of output integrity are reached. For example, assuming a dishonest participant of the
-Atoma Network controls a percentage `r` of the network, the probability, `P`, that a quorum of `N > 0` selected nodes (including at least one of the dishonest participant nodes) is
+In the context of AI inference, BFT methods require heavy compute distributed across many nodes with each node performing the same computations independently. This incurs problems with computational power, latency, and scalability.
 
-`P = r^N`.
+##  Zero Knowledge Machine Learning 
 
-In particular, if `N = 5` nodes are selected and the dishonest participant controls one third of the network, that is `r = 1/3`, the probability above becomes
+Zero knowledge machine learning (ZKML) is a technique to ensure data remains private through zero knowledge proofs (ZKPs). When training the LLM, computations are completed without revealing any of the information provided. 
 
-`P = (1/3)^5 = 0.00411..`
+ZKML is much slower because of the encryption methods used and more computationally intensive. Essentially, compiling a deep neural network into zk circuits for proofs is incredibly expensive and difficult. 
 
-with `N = 10`, the probability becomes
+Given the demand for inference, ZKML is not scalable enough to offer fast and low cost results. 
 
-`P = (1/3)^10 = 1.6935e-05`,
+##  Optimistic Machine Learning (OPML) 
+
+OPML approaches computations assuming everything is correct unless proven otherwise. Systems operate assuming computations are valid which initially is more efficient because it avoids the cost and delays incurred by immediate verification
+
+In order to verify however, OPML has challenge periods which are specific time windows where participants in the system can challenge the validity of a computation and if a challenge is validated, a correction is made to ensure accuracy. 
+
+These challenge periods lead to delays in finalizing computations because they allow time for challenges to be raised and resolved. The delay can vary in length preventing immediate finality. 
+
+Immediate finality ensures that once a computation is confirmed, it is considered complete and valid across the entire system. Challenge periods delay this finality leading to inconsistent states if challenges alter the initial assumptions. The outcome of a computation might change after assuming correctness which disrupts any subsequent operations. 
+
```

Comment:
Great, let's also add a sentence on why OpML isn't an optimal protocol for rewarding nodes, as verifiers need to attest the outputs by computing these, but only get rewards in cases there is a dispute resolved on their behalf